[fix] regex description for routes (#4620)
Kong/docs.konghq.com: src/gateway/how-kong-works/routing-traffic.md

#### Using Regex in paths

For a path to be considered a regular expression, it must be prefixed with a `~`: 

```
paths: ["~/foo/bar$"]
```
Any path that isn't prefixed with a `~` will be considered plain text: 

```
"paths": ["/users/\d+/profile", "/following"]
```

> For a deeper look into how the router processes regular expressions, read [performance considerations when using Expressions](/gateway/latest/key-concepts/routes/expressions/#performance-considerations-when-using-expressions).

Reviewer: ```suggestion
For more information about how the router processes regular expressions, see [performance considerations when using Expressions](/gateway/latest/key-concepts/routes/expressions/#performance-considerations-when-using-expressions).
```
Could just be a personal preference, "deeper look" sounds kind of marketing-y to me

##### Evaluation order

> The router evaluates routes using the `priority` field of the `Route` where a route is configured.

Reviewer: ```suggestion
The router evaluates routes using the `regex_priority` field of the `Route` where a route is configured.
```
Not sure if that's accurate. I only see a `regex_priority` field in the example below, not a `priority` field.

3. `/version/any/`
4. `/version`

> Routers that employ a large amount of regexes can consume traffic intended for other rules.

Reviewer: ```suggestion
Routers with a large number of regexes can consume traffic intended for other rules.
```
Is this a good thing or bad thing? It almost feels like it's missing a "This results in...."

Author: bad thing, should I make it more clear?
